### PR TITLE
continuum_mechanics : added more beam problems

### PR DESCRIPTION
**Pull request by jashan498**
Added exemplar problems : 
```
                                    20KN         8KN
       10 KN/m                      ||           ||
     ||||||||||||||                 \/           \/
      _______________________________________________
     |_______________________________________________|         
           /\                                 OOO

     |-----|------|-----------------|----------|--|--|
        1m    1m          3m              2m   .5m .5m
```
```
                           ramp laod
                       |---------------|
                          12 KN
  \\\\|  4 KN/m            ||
  \\\\||||||||||||         \/
  \\\\|________________________________
  \\\\|________________________________|                              
  \\\\|          :          :          :
  \\\\|----------|-----|----|----------|  

          2.0 m     1m   1m      2.0 m
```
```
                   2 KN/m
              |||||||||||||||||
      ____________________________________________________ 
     O____________________________________________________| o 5KN-m        
    / \                                   /\                

     |--------|----------------|----------|---------------|
         2m           4m            2m            3m
```

#### Release Notes

* physics.continuum_mechanics
  * added more Beam examples.

### CODE REVIEW

File: doc/src/modules/physics/continuum_mechanics/beam_problems.rst [hunk 1]
```diff
@@ -203,3 +203,170 @@ is applied from the mid til the end of the beam.
...
                                                    E⋅I                                                
+
+Example 4
+---------
+
+An overhanging beam of length 8 meters is pinned at 1 meter from starting point
+and supported by a roller 1 meter before the other end. It is subjected
+to a distributed constant load of 10 KN/m from the starting point till
+2 meters away from it. Two pointloads of 20KN and 8KN are applied at
+5 meters and 7.5 meters away from the starting point respectively.
```
Comment (moorepants) [inline doc/src/modules/physics/continuum_mechanics/beam_problems.rst]: Did you come up with this wording or is it copied from a book, etc?

File: doc/src/modules/physics/continuum_mechanics/beam_problems.rst [hunk 2]
```diff
@@ -203,3 +203,170 @@ is applied from the mid til the end of the beam.
...
+
+.. note::
+
+    Using the sign convention of upward forces and clockwise moment
+    being positive.
+
+>>> from sympy.physics.continuum_mechanics.beam import Beam
```
Comment (moorepants) [inline doc/src/modules/physics/continuum_mechanics/beam_problems.rst]: These should be in in `.. code::` blocks I believe.

File: doc/src/modules/physics/continuum_mechanics/beam_problems.rst (hunk 1 above)
Comment (jashan498) [reply doc/src/modules/physics/continuum_mechanics/beam_problems.rst]: These were specific questions like "Find max deflection in the beam " along with a diagram given. 
So yeah, I wrote the wording on my own.

**Comment by Abdullahjavednesar** [comment]
ping @moorepants

File: doc/src/modules/physics/continuum_mechanics/beam_problems.rst (hunk 1 above)
Comment (moorepants) [reply doc/src/modules/physics/continuum_mechanics/beam_problems.rst]: Ok, that's good then, we just want to abide by copyright rules.

File: doc/src/modules/physics/continuum_mechanics/beam_problems.rst [hunk 3]
```diff
@@ -56,45 +56,46 @@ of 12 kN is applied at the end.
     Since a user is free to choose its own sign convention we are considering
     the upward forces and clockwise bending moment being positive.
 
+.. code::
```
Comment (moorepants) [inline doc/src/modules/physics/continuum_mechanics/beam_problems.rst]: I think you can use `.. code:: pycon` for correct syntax highlighting if it doesn't do it correctly by default.

**Review by moorepants — CHANGES_REQUESTED** [review]
This looks fine. Please add in your ASCII art beams into the rst files if possible. That really helps the reader understand the problem.

**Comment by parsoyaarihant** [comment]
@jashan498  ping.

File: doc/src/modules/physics/continuum_mechanics/beam_problems.rst [hunk 4]
```diff
@@ -106,6 +116,13 @@ point. There are two simple supports below the beam. One at the end
 and another one at a distance of 10 meters from the start. The
 deflection is restricted at both the supports.
 
+
+  || 8 N                                       120 Nm
+  \/______________________________________________|
+  |_______________________________________________|
+              /\                                 /\
+  |------------|---------------------------------|
```
Comment (moorepants) [inline doc/src/modules/physics/continuum_mechanics/beam_problems.rst]: Do these render correctly when you build the docs?

File: doc/src/modules/physics/continuum_mechanics/beam_problems.rst [hunk 5]
```diff
@@ -154,7 +171,16 @@ Example 3
 A beam of length 6 meters is having a roller support at the start and a
 hinged support at the end. A clockwise moment of 1.5 kN-m is applied at the mid
 of the beam. A constant distributed load of 3 kN/m and a ramp load of 1 kN/m
-is applied from the mid til the end of the beam.
+is applied from the mid till the end of the beam.
+
+                              ramp laod = 1 KN/m
+                            constant laod = 3 KN/m
```
Comment (moorepants) [inline doc/src/modules/physics/continuum_mechanics/beam_problems.rst]: laod > load

File: doc/src/modules/physics/continuum_mechanics/beam_problems.rst (hunk 4 above)
Comment (moorepants) [reply doc/src/modules/physics/continuum_mechanics/beam_problems.rst]: I get this error:

```

Warning, treated as error:
/home/moorepants/src/sympy/doc/src/modules/physics/continuum_mechanics/beam_problems.rst:56:Block quote ends without a blank line; unexpected unindent.
Makefile:49: recipe for target 'html' failed
make: *** [html] Error 2

```

Make sure to build the docs locally.

Comment (jashan498) [reply doc/src/modules/physics/continuum_mechanics/beam_problems.rst]: It is giving `Undefined substitution` error at the start of every ascii figure:
```
/home/jashan/sympy/doc/src/modules/physics/continuum_mechanics/beam_problems.rst:55:Undefined substitution referenced: "---------------".
```
is `..parsed-literal::` correct block for these type of figures?

File: doc/src/modules/physics/continuum_mechanics/beam_problems.rst [hunk 6]
```diff
@@ -203,3 +235,213 @@ is applied from the mid til the end of the beam.
...
+5 meters and 7.5 meters away from the starting point respectively.
+
+::
+
+                                 20KN         8KN
+    10 KN/m                      ||           ||
+  ||||||||||||||                 \/           \/
+   _______________________________________________
+  |_______________________________________________|
+        /\                                  OOO
```
Comment (moorepants) [inline doc/src/modules/physics/continuum_mechanics/beam_problems.rst]: Maybe roller: 
```
 /\
oooo
```
or just a single `o`.

File: doc/src/modules/physics/continuum_mechanics/beam_problems.rst [hunk 7]
```diff
@@ -203,3 +235,213 @@ is applied from the mid til the end of the beam.
...
+the beam. A point load of 12KN is also applied in same direction 4 meters
+away from start.
+
+::
+
+                       ramp load = 1 KN/m
+                       |---------------|
+                          12 KN
+  \\\\|  4 KN/m            ||
+  \\\\||||||||||||         \/
```
Comment (moorepants) [inline doc/src/modules/physics/continuum_mechanics/beam_problems.rst]: I think you can overlap the ramp load at the point load. Also you can probably make it look like a ramp.

**Comment by moorepants** [comment]
Please address my latest comments and I think we can merge then. Thanks.

**Comment by jksuom** [comment]
There were two that hadn't even started and one with connection failure. That is unusual.

**Review by moorepants — APPROVED** [review]

**Pull request merged**